### ISSUE TITLE
ConcurrentModificationException in GameState.setup() when component setup() adds components

### ISSUE THREAD

**Issue by uckelman** [vassalengine/vassal#3969]
| Bugzilla |  |
|----------|----------|
| ID | 14456 |
| Reported | 2021-05-07 16:25:00 +0200 |
| Modified | 2021-05-10 20:54:56 +0200 |
| Product | VASSAL |
| Component | Player |
| Version | 3.5.5 |
| Platform | All |
| OS | All |
| Status | RESOLVED |
| Resolution | FIXED |
| Priority | unspecified |
| Severity | normal |
| Milestone | 3.5.6 |
| Reporter | [Joel Uckelman](mailto:uckelman@nomic.net) |
| Assignee | [Bugs](mailto:bugs@vassalengine.org) |
| CC | brian@brianreynolds.net |

| Attachments | |
|-------|-------|
| [errorLog](http://www.vassalengine.org/files/bz/12965) | [Joel Uckelman](mailto:uckelman@nomic.net) 2021-05-07 16:25:00 +0200 text/plain 41989 the errorLog |

#### [Joel Uckelman](mailto:uckelman@nomic.net) 2021-05-07 16:25:33 +0200
<pre>josh.derksen@gmail.com

in edit module mode, clicked new game. Prior to this, I edited a game board. It had a rectangular grid, then I added a multi-zone grid with a child zone, then cut and pasted the rectangular grid into it.

Lynnvander_08_BC v1.0 3.5.5

java.util.ConcurrentModificationException: null
 at java.base/java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1013)
 at java.base/java.util.ArrayList$Itr.next(ArrayList.java:967)
 at VASSAL.build.module.GameState.setup(GameState.java:396)
 at VASSAL.build.module.GameState$6.actionPerformed(GameState.java:198)</pre>

#### [Joel Uckelman](mailto:uckelman@nomic.net) 2021-05-07 16:25:34 +0200
<pre>Created attachment 12965
the errorLog</pre>

#### [Brian Reynolds](mailto:brian@brianreynolds.net) 2021-05-10 18:52:10 +0200
<pre>*** Bug 14458 has been marked as a duplicate of this bug. ***</pre>

#### [Brian Reynolds](mailto:brian@brianreynolds.net) 2021-05-10 18:52:22 +0200
<pre>*** Bug 14457 has been marked as a duplicate of this bug. ***</pre>

#### [Brian Reynolds](mailto:brian@brianreynolds.net) 2021-05-10 18:53:11 +0200
<pre>Submitted PR to copy the list before starting the loop.

Emailed user for copy of module just so I can "explore and investigate" if there is any deeper misbehavior going on.</pre>
